cmd/argocd/commands/login.go

```diff
@@ -38,35 +47,47 @@ func NewLoginCommand(globalClientOpts *argocdclient.ClientOptions) *cobra.Comman
 			errors.CheckError(err)
 			if !tlsTestResult.TLS {
 				if !globalClientOpts.PlainText {
-					askToProceed("WARNING: server is not configured with TLS. Proceed (y/n)? ")
+					cli.AskToProceed("WARNING: server is not configured with TLS. Proceed (y/n)? ")
 					globalClientOpts.PlainText = true
 				}
 			} else if tlsTestResult.InsecureErr != nil {
 				if !globalClientOpts.Insecure {
-					askToProceed(fmt.Sprintf("WARNING: server certificate had error: %s. Proceed insecurely (y/n)? ", tlsTestResult.InsecureErr))
+					cli.AskToProceed(fmt.Sprintf("WARNING: server certificate had error: %s. Proceed insecurely (y/n)? ", tlsTestResult.InsecureErr))
 					globalClientOpts.Insecure = true
 				}
 			}
-
-			username, password = cli.PromptCredentials(username, password)
-
 			clientOpts := argocdclient.ClientOptions{
 				ConfigPath: "",
 				ServerAddr: server,
 				Insecure:   globalClientOpts.Insecure,
 				PlainText:  globalClientOpts.PlainText,
 			}
-			conn, sessionIf := argocdclient.NewClientOrDie(&clientOpts).NewSessionClientOrDie()
-			defer util.Close(conn)
+			acdClient := argocdclient.NewClientOrDie(&clientOpts)
+			setConn, setIf := acdClient.NewSettingsClientOrDie()
+			defer util.Close(setConn)
+
+			acdSet, err := setIf.Get(context.Background(), &settings.SettingsQuery{})
+			errors.CheckError(err)
+
+			ctxName = cli.PromptMessage("Enter a name for this context", ctxName)
+			username = cli.PromptUsername(username)
+
+			// Perform the login
+			var tokenString string
+			if username == "admin" || !ssoConfigured(acdSet) {
```

Comment:
From experience, hardcoding usernames into code is a sure path to madness, not to mention deep-seated bugs down the road.  Can we maybe use a less-eldritch method such as setting up a constant or global list of valid administrator usernames if we're going to do it this way, or better yet find some way to check the K8s secrets for what username is set for the  @administrator?